sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py

```diff
@@ -230,6 +230,11 @@ async def create_container(
         if offer_throughput is not None:
             request_options["offerThroughput"] = offer_throughput
 
+        auto_scale_settings = kwargs.pop('auto_scale_settings', None)
+        if auto_scale_settings is not None:
+            auto_scale_settings = auto_scale_header(auto_scale=auto_scale_settings)
+            request_options["autoUpgradePolicy"] = auto_scale_settings
```

Comment:
It feels like this should be one step
```Python
if auto_scale_settings is not None:
  request_options["autoUpgradePolicy"] = str(auto_scale_settings.__dict__)
```

The `auto_scale_method` doesn't feel necessary.